### PR TITLE
Support for BGP Graceful Restart Notifications Extensions as per RFC8538

### PR DESCRIPTION
**Pull request by apratimmukherjee**
Redocly View: [ReDoc Interactive Demo (redocly.github.io)](https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/open-traffic-generator/models/bgp_gr_notification/artifacts/openapi.yaml&nocors#tag/Configuration) and PR: https://github.com/open-traffic-generator/models/pull/383

To use this model to generate example snappi tests.
go get github.com/open-traffic-generator/snappi/gosnappi@bgp_gr_notifications

Changes: 
1) In set_config , new optional attribute 'enable_notification' is added to devices/bgp/.../peers/graceful_restart with default true.
2) In control_action / bgp / initiate_graceful_restart, option added to send a Notification when peer is going down. 

Example usage: 
```

//Config creation 
{
	...
	config := gosnappi.NewConfig()

    // add ports
    p1 := config.Ports().Add().SetName("p1").SetLocation(<<location>)
        
    // add devices
    d1 := config.Devices().Add().SetName("p1d1")    
    d1Eth1 := d1.Ethernets().
                Add().
                SetName("p1d1eth").
                SetMac("00:00:01:01:01:01").
                SetMtu(1500)

    d1Eth1.Connection().SetPortName(p1.Name())

    d1Eth1.Ipv4Addresses().
                Add().
                SetName("p1d1ipv4").
                SetAddress("1.1.1.2").
                SetGateway("1.1.1.1").
                SetPrefix(24)

    d1Bgp := d1.Bgp().
                SetRouterId("1.1.1.2")

    d1BgpIpv4Interface1 := d1Bgp.
                Ipv4Interfaces().Add().
                SetIpv4Name("p1d1ipv4")

    d1BgpIpv4Interface1Peer1 := d1BgpIpv4Interface1.
                Peers().
                Add().
                SetAsNumber(2222).
                SetAsType(gosnappi.BgpV4PeerAsType.EBGP).
                SetPeerAddress("1.1.1.1").
                SetName("p1d1bgpv4")

    // Enable GR 
    d1BgpIpv4Interface1Peer1.GracefulRestart().
                SetEnableGr(true).
                SetRestartTime(60).
		SetEnableNotification(true) //Default true : Control to enable/disable Notification extension in BGP Capability                 

    d1BgpIpv4Interface1Peer1V4Route1 := d1BgpIpv4Interface1Peer1.
                V4Routes().
                Add().
                SetName("p1d1peer1rrv4")

    d1BgpIpv4Interface1Peer1V4Route1.Addresses().Add().
                SetAddress("10.10.10.1").
                SetPrefix(32).
                SetCount(4).
                SetStep(1)

    d1BgpIpv4Interface1Peer1V4Route1.Advanced().
                SetMultiExitDiscriminator(50).
                SetOrigin(gosnappi.BgpRouteAdvancedOrigin.EGP)
 
    d1BgpIpv4Interface1Peer1V4Route1AsPath := d1BgpIpv4Interface1Peer1V4Route1.AsPath().
                SetAsSetMode(gosnappi.BgpAsPathAsSetMode.INCLUDE_AS_SET)

    d1BgpIpv4Interface1Peer1V4Route1AsPath.Segments().Add().
                SetAsNumbers([]uint32{1112, 1113}).
                SetType(gosnappi.BgpAsPathSegmentType.AS_SEQ)
}

To send trigger with Notification enabled:
{
...
    grAction := gosnappi.NewControlAction()
    bgpPeersRestart := grAction.Protocol().
                                Bgp().
                                InitiateGracefulRestart()
    peerNames := []string{"p1d1bgpv4"}
    bgpPeersRestart.
            SetPeerNames(peerNames).
            SetRestartDelay(20)
    
    if sendNotification == true {
        notification:= bgpPeersRestart.Notification()
        if sendHardReset == true {    
            notification.Cease().SetSubcode(gosnappi.DeviceBgpCeaseErrorSubcode.HARD_RESET_CODE6_SUBCODE9)
        } else {
            /* Send anything else except hard reset */
            notification.Cease().SetSubcode(gosnappi.DeviceBgpCeaseErrorSubcode.OUT_OF_RESOURCES_CODE6_SUBCODE8)
        }
    }
        
    if _, err := client.Api().SetControlAction(grAction); err != nil {
        t.Fatal(err)
    }
...
}

```

### CODE REVIEW

File: control/bgp.yaml [hunk 1]
```diff
@@ -116,4 +120,65 @@ components:
           format: uint32
           maximum: 3600
           default: 30
-          x-field-uid: 2
\ No newline at end of file
+          x-field-uid: 2
```
Comment (SuryyaKrJana) [inline control/bgp.yaml:123]: so we are introducing convention of "spaces" for better readability?

File: control/bgp.yaml [hunk 2]
```diff
@@ -116,4 +120,65 @@ components:
...
+              x-field-uid: 3
+            update_message_error:
+              x-field-uid: 4
+            hold_timer_expired:
+              x-field-uid: 5
+            finite_state_machine_error:
+              x-field-uid: 6
+            custom:
+              x-field-uid: 7
+
```
Comment (SuryyaKrJana) [inline control/bgp.yaml:163]: same as previous comment related to space

**Review by SuryyaKrJana — APPROVED** [review]

File: control/bgp.yaml (hunk 2 above)
Comment (apratimmukherjee) [reply control/bgp.yaml:163]: Want to keep space between diff. objects.

File: control/bgp.yaml (hunk 1 above)
Comment (apratimmukherjee) [reply control/bgp.yaml:123]: There is no extra space here (checked codebase ) . This shows up due to previous eof getting removed on lhs.

**Pull request merged**